iothub_client/samples/iothub_ll_c2d_sample/iothub_ll_c2d_sample.c

```diff
@@ -103,13 +143,21 @@ static IOTHUBMESSAGE_DISPOSITION_RESULT receive_msg_callback(IOTHUB_MESSAGE_HAND
     }
     g_message_recv_count++;
 
+#ifdef USE_C2D_ASYNC_ACK
+    (void)singlylinkedlist_add(g_cloudMessages, message);
+    return IOTHUBMESSAGE_ASYNC_ACK;
```

Comment:
I see these are still open but @ewertons please incorporate them (or re-wordsmith if needed), otherwise I think this will be tricky for folks to figure out.

Reply:
I just added that to the changes. Thanks John.